doc/syntax.rst

```diff
@@ -137,6 +137,88 @@ For a clear example refer to the rendered example
 :ref:`sphx_glr_tutorials_plot_parse.py` and compare it to the generated
 :download:`original python script <tutorials/plot_parse.py>`
 
+.. _non_python_source:
+
+Examples in other programming languages
+=======================================
+
+Sphinx-Gallery also supports rendering HTML pages for examples written in programming
+languages other than Python, although these examples are not currently executed or
+scanned for output. See :ref:`filename/ignore patterns <build_pattern>` for
+configuration settings.
+
+For such examples, the header for the example is defined by the first comment block
+in the file, which must contain a reST title, and may contain any additional reST
+content that should appear above the example. For example, A C++ example could start
+with:
+
+.. code:: C++
+
+  // My Awesome Example
+  // ==================
+  //
+  // The description continues as long as there are lines
+  // that start with a comment character.
+
+reST content can likewise be embedded in comments that are marked with a special
+delimiter, where that delimiter depends on the comment characters used by the language
+of the example. Valid special delimiters are:
+
+1. The comment character followed by ``%%``. For example ``//%%`` for C++.
+2. The comment character followed by a space, followed by ``%%``. For example, ``// %%``
+   for C++.
+3. A line of at least 20 comment characters. For example, ``////////////////////`` for
+   C++.
+
+Any text following the special delimiter on the same line will be converted into a reST
+heading (underlined with ``-``). The reST block continues until a line that does not
```

Comment:
It means that the underlining will be added, using the `-` character. I specified this so the user would know what character to use if they wanted other headings at either the same or a different level.

The existing header syntax is also supported. However, in the case of Matlab, which has had the notion of "code cells" for a long time (possibly where this idea originated), the style is that text appearing on the same line as the `%%` delimiter is treated as a heading, rather than ignored as sphinx-gallery currently does for Python scripts. Beyond Matlab, I thought it was a handy syntax especially in the case where all you want to add is the heading.

Reply:
Ah yes, sorry I think I got confused.